includes/class-wc-stripe-payment-tokens.php

```diff
@@ -364,7 +374,11 @@ public function woocommerce_payment_token_deleted( $token_id, $token ) {
 		$stripe_customer = new WC_Stripe_Customer( get_current_user_id() );
 		if ( WC_Stripe_Feature_Flags::is_upe_checkout_enabled() ) {
 			if ( WC_Stripe_UPE_Payment_Gateway::ID === $token->get_gateway_id() ) {
-				$stripe_customer->detach_payment_method( $token->get_token() );
+				try {
```

Comment:
How about also having a `try` block to handle $stripe_customer->delete_source() below (https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2668/files#diff-f80280bf89ae5350f2729f8c88d0d39cd4df34c7f13fe87db1445bcb03d6ac34R385) too? Looks like it can throw an Exception as well.

Reply:
Good catch :+1: